tests/ludwig/augmentation/test_augmentation_pipeline.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+import torch
+
+from ludwig.features.image_feature import AugmentationPipeline
+
+
+# define fixture for test  image augmentation
+@pytest.fixture(scope="module")
+def test_image():
+    # return random normal batch of images of size 2x3x32x32 [batch_size, channels, height, width]
+    return torch.randn(2, 3, 32, 32)
+
+
+@pytest.mark.parametrize(
+    "augmentation_pipeline_ops",
+    [
+        [],
```

Comment:
Is this empty list the one that causes test failures?

Reply:
The empty list does not cause an error.  

I assuming that if **user does not** specify `augmentation` in the config, then after all the default setting and backward compatiblity  processing the generated configuration dictionary will have an `augmentation` section added with an empty list.

Let me know if I have the wrong understanding.

Reply:
Gotcha. Yes that is the correct understanding!

To close this loop, I would like if you could add tests that explicitly check if an error is raised if a user manually sets `augmentation: None` (note that this is subtly different from the user choosing _not_ to specify `augmentation` at all) or to some erroneous (non-list) type. You could put maybe one here to see that the error is surfaced through to the pipeline logic layer and then the rest under `tests/schema` where you can use `validate_config` to do some basic schema validation tests.

Reply:
with `augmentation` set to `None` in the config file, this exception is raised
```
File "/usr/local/lib/python3.9/site-packages/jsonschema/validators.py", line 1058, in validate
    raise error
jsonschema.exceptions.ValidationError: None is not of type 'array'
```

this should address the comment.

Reply:
great!